### PR TITLE
Document Azure usage

### LINKED ISSUE #797: Azure documentation
### What type of suggestions?

Missing Information

### What page is concerned?

_No response_

### What version of the docs?

_No response_

### What is the issue?

Since @mrsimonemms is implementing Azure, I want this to be on our mind. Not sure of the progress or ETA of this new cloud though.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mrsimonemms**
## Description

Add the documentation for Azure. As with everything, this has been based on the Civo stuff but Azurified.

Couple of notes:
- I would favour publishing Azure when we release v2.8 docs not v2.7. It's in from `2.7.4`, so could be confusing if we get support requests for `2.7.0~3`
- there are some [outstanding images required](https://github.com/konstructio/kubefirst/issues/2320). This shouldn't be published until they're added to this PR

As always, happy to do demos if required.

## Related Issue(s)

Fixes #797

## How to test

Run `npm start` and read things under the "Azure" tab.

### CODE REVIEW

**Comment by fharper** [comment]
Azure docs shouldn't be in 2.7 unless it's already release in 2.7.x. For now, it should only reside in v.next (the `docs` folder, not in any versioned_docs). If we release azure in 2.7.x, we will do what the last commit to this PR does, but if it's in a 2.8.X release, we will create a new docs version for 2.8, and automatically it will have Azure content as the new version for docs are a copy of v.next. Happy to discuss more if it's not clear. I created an issue to clarify how and where to add new feature docs https://github.com/konstructio/kubefirst-docs/issues/820 as it's not clear if you don't own the docs, sorry

File: docs/azure/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+<div class="cloud_logo">
+  <AzureLogo width="200px" alt="Azure Logo" />
+</div>
+
+# Overview
+
+:::caution
+
+Azure is in beta. Use at your own risks.
```
Comment (konstructcarrie) [inline docs/azure/overview.mdx]:
```suggestion
Azure is in beta. Use at your own risk.
```

File: docs/azure/partials/common/_prerequisites.mdx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+### Azure Prerequisites
+
+For kubefirst to be able to provision your Azure cloud resources:
```
Comment (konstructcarrie) [inline docs/azure/partials/common/_prerequisites.mdx:3]:
```suggestion
For Kubefirst to be able to provision your Azure cloud resources:
```

File: versioned_docs/version-2.7/azure/overview.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+<div class="cloud_logo">
+  <AzureLogo width="200px" alt="Azure Logo" />
+</div>
+
+# Overview
+
+:::caution
+
+Azure is in beta. Use at your own risks.
```
Comment (konstructcarrie) [inline versioned_docs/version-2.7/azure/overview.mdx]:
```suggestion
Azure is in beta. Use at your own risk.
```

File: versioned_docs/version-2.7/azure/overview.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+</Tabs>
+
+## Known Limitations
+
+<KnownLimitations />
+
+## Azure Specific
+
+- Azure is still in beta, but should be usable in production
+- Workload cluster are not available yet
```
Comment (konstructcarrie) [inline versioned_docs/version-2.7/azure/overview.mdx]:
```suggestion
- Workload clusters are not available yet
```

File: versioned_docs/version-2.7/azure/partials/common/_prerequisites.mdx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+### Azure Prerequisites
+
+For kubefirst to be able to provision your Azure cloud resources:
```
Comment (konstructcarrie) [inline versioned_docs/version-2.7/azure/partials/common/_prerequisites.mdx]:
```suggestion
For Kubefirst to be able to provision your Azure cloud resources:
```

**Comment by konstructcarrie** [comment]
@mrsimonemms Added some edits for ticky-tack language stuff. Otherwise LGTM 👍

File: docs/azure/partials/common/_prerequisites.mdx (hunk 2 above)
Comment (mrsimonemms) [reply docs/azure/partials/common/_prerequisites.mdx:3]: Whilst I agree that it should be a Kapital K, every other instance is lowerkase k. I've kept that for consistency, but I think we should perhaps update that outside of this PR.

File: versioned_docs/version-2.7/azure/partials/common/_prerequisites.mdx (hunk 5 above)
Comment (mrsimonemms) [reply versioned_docs/version-2.7/azure/partials/common/_prerequisites.mdx]: Anything under the `versioned_docs` folder is autogenerated by the `/docs` folder

File: docs/azure/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+</Tabs>
+
+## Known Limitations
+
+<KnownLimitations />
+
+## Azure Specific
+
+- Azure is still in beta, but should be usable in production
+- Workload cluster are not available yet
```
Comment (mrsimonemms) [inline docs/azure/overview.mdx]:
```suggestion
- Workload clusters are not available yet
```

File: versioned_docs/version-2.7/azure/overview.mdx (hunk 4 above)
Comment (mrsimonemms) [reply versioned_docs/version-2.7/azure/overview.mdx]: Fixed in docs

File: docs/azure/partials/common/_prerequisites.mdx (hunk 2 above)
Comment (mrsimonemms) [reply docs/azure/partials/common/_prerequisites.mdx:3]: Will do as part of #773

**Review by konstructcarrie — DISMISSED** [review]
Approved (with the caveat that this is merged into the versioning that @fharper suggested) I believe this is in the right "place"

**Review by konstructcarrie — APPROVED** [review]

**Pull request merged**